### ISSUE TITLE
Concurrent map writes error in RegisterBodyEncoder

### ISSUE THREAD

**Issue by natehart** [getkin/kin-openapi#917]
The [bodyEncoders](https://github.com/getkin/kin-openapi/blob/master/openapi3filter/req_resp_encoder.go#L21) map is not thread safe. If encoders are registered in parallel, [this line](https://github.com/getkin/kin-openapi/blob/672dc3233232be7cb1485f3b87bcae981693bd91/openapi3filter/req_resp_encoder.go#L32C2-L32C14) can cause a fatal "concurrent map writes" error. I discovered this due to having parallel tests that each used a new instance of my rest server.

This is likely low priority as this should't really ever happen in production code, but it's probably still worth making this map thread safe by adding a read-write mutex to guard all functions that access that map.

**Comment by fenollp** [comment]
A `sync.RWMutex` should do it indeed
